src/lib/expandApplyAtRules.js

```diff
@@ -432,6 +432,23 @@ function processApply(root, context, localCache) {
 
       let rules = applyClassCache.get(applyCandidate)
 
+      // Verify that we can apply the class
+      for (let [, rule] of rules) {
+        if (rule.type === 'atrule') {
+          continue
+        }
+
+        rule.walkRules(() => {
+          throw apply.error(
+            [
+              `The \`${applyCandidate}\` class cannot be applied because it uses nested CSS.`,
+              'Please enable a CSS nesting plugin *before* Tailwind in your configuration.',
+              'See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting',
+            ].join('\n')
+          )
+        })
+      }
```

Comment:
What if the applied thing is nested _inside_ another rule? Can that happen here? If so we should probably walk rule.parent all the way to the root to detect that.

Reply:
You can apply user CSS and that is broken:
1. https://play.tailwindcss.com/MevI8qxvI6?file=css
2. https://play.tailwindcss.com/sdpYgh5nZS?file=css

Reply:
Aha I see what you mean, this should also be solved by this PR already (added an additional test to proof it). The reason why it works I think is because if you have `.foo .bar {}` we register both `foo` and `bar` as applyable candidates pointing to the same node (or a clone)

So if you have `.foo { .bar {} }` both `foo` and `bar` are applyable and point to the same node. I'm not 100% sure on this (my brain is tired lol) but the test I added seems to prove this as well 🤔